### DISCUSSION TITLE
Switch Port Topology

### DISCUSSION THREAD

**Discussion by opensourcefan** [mattieserver/netbox-topology-views#316] [Ideas]
Don't know how hard or easy this would be but how about displaying the ports of a switch with labels or hover labels. 

For example, for a 48 port switch, it could display two rows with 24 x network-socket.svg on each row. They could be labeled to whatever they're connected to. 

I suppose the only difference to currently selecting a single switch to display would be the all the ports would be visible and they would be arranged horizontally like the actual device.

Does this explanation make sense?

**Comment by opensourcefan** [comment]
something like this maybe: sorry for the crudeness :)

![Screenshot 2023-05-29 203903](https://github.com/mattieserver/netbox-topology-views/assets/10550038/76aa882d-3dc9-40ec-a5ca-886d9a94a214)

**Comment by dreng** [comment]
Interesting idea. A long time ago I saw a screenshot of "Pathfinder" made by [Tripunkt](https://www.pathfinder.de/en/). There, the devices are shown in racks and the ports are shown connected to each other. This looks great, but it is not flexible enough, because devices can have ports on both sides and devices that are not built into a rack are not taken into account.

Your idea could be a good compromise. We just have to be careful that it doesn't end up in a mess. It definitely has to be an option, because with this representation several thousand icons can quickly come together. Implementing the feature shouldn't even mean much of a change. However, if we want to offer both, the effort could be high.

To avoid it getting completely out of hand, I would not automatically group the icons together and not provide them with a title. The placement of the icons should still be calculated by vis-network or overwritten manually.
